kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/KafkaProtocolFilterReconcilerTest.java

```diff
@@ -0,0 +1,140 @@
+
+import io.kroxylicious.kubernetes.filter.api.v1alpha1.KafkaProtocolFilter;
+import io.kroxylicious.kubernetes.filter.api.v1alpha1.KafkaProtocolFilterBuilder;
+import io.kroxylicious.kubernetes.operator.assertj.KafkaProtocolFilterStatusAssert;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class KafkaProtocolFilterReconcilerTest {
+
+    // @formatter:off
+    public static final KafkaProtocolFilter FILTER = new KafkaProtocolFilterBuilder()
+            .withNewMetadata()
+                .withName("foo")
+                .withGeneration(42L)
+            .endMetadata()
+            .withNewSpec()
+                .withType("org.example.MyFilter")
+                .withConfigTemplate(Map.of(
+                        "normalProp", "normalValue",
+                        "securePropA", "${secret:my-secret:key}",
+                        "securePropB", "${configmap:my-configmap:key}"))
+            .endSpec()
+            .build();
+
+    public static final Secret SECRET = new SecretBuilder()
+            .withNewMetadata()
+            .withName("my-secret")
+            .withGeneration(42L)
+            .endMetadata()
+            .addToData("key", "value")
+            .build();
+
+    public static final ConfigMap CONFIG_MAP = new ConfigMapBuilder()
+            .withNewMetadata()
+            .withName("my-configmap")
+            .withGeneration(42L)
+            .endMetadata()
+            .addToData("key", "value")
+            .build();
+    // @formatter:on
+
+    @Test
+    void shouldSetResolvedRefsToFalseWhenReferentsNotFound() {
+        // given
+        Clock z = Clock.fixed(Instant.EPOCH, ZoneId.of("Z"));
+        var reconciler = new KafkaProtocolFilterReconciler(z, SecureConfigInterpolator.DEFAULT_INTERPOLATOR);
+
+        Context<KafkaProtocolFilter> context = mock(Context.class);
+        when(context.getSecondaryResourcesAsStream(Secret.class)).thenReturn(Stream.of());
+        when(context.getSecondaryResourcesAsStream(ConfigMap.class)).thenReturn(Stream.of());
+
+        // when
+        var update = reconciler.reconcile(FILTER, context);
+
+        // then
+        assertThat(update).isNotNull();
+        assertThat(update.isPatchStatus()).isTrue();
+        assertThat(update.getResource()).isPresent();
+        KafkaProtocolFilterStatusAssert.assertThat(update.getResource().get().getStatus())
+                .hasObservedGenerationInSyncWithMetadataOf(FILTER)
+                .singleCondition()
+                .hasObservedGenerationInSyncWithMetadataOf(FILTER)
+                .isResolvedRefsFalse("MissingInterpolationReferences", "Referenced Secrets [my-secret] ConfigMaps [my-configmap] not found")
+                .hasLastTransitionTime(ZonedDateTime.ofInstant(z.instant(), z.getZone()));
+
+    }
+
+    @Test
+    void shouldSetResolvedRefsToTrueWhenReferentsFound() {
```

Comment:
Think it's worth a check for each referent in isolation, maybe you could parameterize the test so we're checking:
- secret resolved, configmap not resolved
- both unresolved
- configmap resolved, secretmap not resolved

Reply:
Done